daemon/cmd/daemon_main.go

```diff
@@ -1534,20 +1537,17 @@ func initEnv(cmd *cobra.Command) {
 		if !option.Config.EnableIPv4 {
 			option.Config.EnableIPv4FragmentsTracking = false
 		} else {
-			supportedMapTypes := probes.NewProbeManager().GetMapTypes()
-			if !supportedMapTypes.HaveLruHashMapType {
+			if err := features.HaveMapType(ebpf.LRUHash); err != nil {
 				option.Config.EnableIPv4FragmentsTracking = false
 				log.Info("Disabled support for IPv4 fragments due to missing kernel support for BPF LRU maps")
 			}
 		}
 	}
 
 	if option.Config.EnableBPFTProxy {
-		if h := probes.NewProbeManager().GetHelpers("sched_act"); h != nil {
-			if _, ok := h["bpf_sk_assign"]; !ok {
-				option.Config.EnableBPFTProxy = false
-				log.Info("Disabled support for BPF TProxy due to missing kernel support for socket assign (Linux 5.7 or later)")
-			}
+		if err := features.HaveProgramHelper(ebpf.SchedACT, asm.FnSkAssign); err != nil {
```

Comment:
Small nit: I would use SchedCLS as well here, since we don't use act_bpf internally, so we also don't need to add unnecessary dependency.